statsmodels/base/_screening.py

```diff
@@ -0,0 +1,214 @@
+            #print(mcs[:10])
+
+            threshold = mcs[max((self.k_max_add, k0 + self.k_add))]
+            #idx = np.concatenate((idx_nonzero, np.nonzero(mom_cond > threshold)[0] + 1))
+            idx = np.concatenate((idx_nonzero, idx_excl[mom_cond > threshold]))
+            start_params2 = np.zeros(len(idx))
+            start_params2[:len(start_params)] = start_params
+
+            res_pen = model_class(endog, x[:, idx], penal=self.penal,
+                                       pen_weight=nobs * 10,
+                                       **self.init_kwds).fit(method='bfgs',
+                                                start_params=start_params2,
+                                                warn_convergence=False, disp=disp,
+                                                skip_hessian=True)
+
+            keep = np.abs(res_pen.params) > self.threshold_trim
+            # use largest params to keep
+            if len(res_pen.params) > self.k_max_included:
+                # TODO we can use now np.partition with partial sort
+                thresh_params = np.sort(np.abs(res_pen.params))[-self.k_max_included]
+                keep2 = np.abs(res_pen.params) > thresh_params
+                keep = np.logical_and(keep, keep2)
+            idx_nonzero = idx[keep]
+            # TODO: problem with order between using mask and indexing
+            #idx_nonzero.sort()
+
+            if disp:
+                print(idx_nonzero)
+            x0 = x[:, idx_nonzero]
+            start_params = res_pen.params[keep]
+
+            # use mask to get excluded
+            mask_excl = np.ones(k_vars, dtype=bool)
+            mask_excl[idx_nonzero] = False
+            idx_excl = np.nonzero(mask_excl)[0]
+            x1 = x[:, idx_excl]
+            history['idx_nonzero'].append(idx_nonzero)
+            history['keep'].append(keep)
+            history['params_keep'].append(start_params)
+
+        # final esimate
+        res_final = model_class(endog, x[:, idx_nonzero], penal=self.penal,
+                                pen_weight=nobs * 10,
+                                **self.init_kwds).fit(method='bfgs',
+                                          start_params=res_pen.params[keep],
+                                          warn_convergence=False, disp=disp)
+
+        res = ScreeningResults(self,
+                               results_pen = res_pen,
+                               results_final = res_final,
+                               idx_nonzero = idx_nonzero,
+                               idx_excl = idx_excl,
+                               start_params = start_params,
+                               history = history)
+        return res
+
+    def screen_exog_iterator(self, exog_iterator):
+        res_batches = []
+        res_idx = []
+        exog_winner = []
+        exog_idx = []
+        for ex in exog_iterator:
+            res_screen = self.screen_exog(ex, maxiter=20)
+            res_batches.append(res_screen)
```

Comment:
We should not keep res_screen around, too much memory, unless it's a user option.
some history was added for debugging.